tests/src/test/scala/ch/epfl/bluebrain/nexus/tests/kg/files/FilesDsl.scala

```diff
@@ -0,0 +1,85 @@
+package ch.epfl.bluebrain.nexus.tests.kg.files
+
+import akka.http.scaladsl.coding.Coders
+import akka.http.scaladsl.model._
+import akka.http.scaladsl.model.headers._
+import akka.stream.Materializer
+import akka.util.ByteString
+import cats.effect.IO
+import ch.epfl.bluebrain.nexus.tests.Identity.storages.Coyote
+import ch.epfl.bluebrain.nexus.tests.kg.files.model.FileInput
+import ch.epfl.bluebrain.nexus.tests.{CirceUnmarshalling, HttpClient}
+import io.circe.Json
+import org.apache.commons.codec.Charsets
+import org.scalatest._
+import org.scalatest.concurrent.ScalaFutures
+import org.scalatest.matchers.should.Matchers
+
+import java.util.Base64
+import scala.concurrent.ExecutionContext
+
+class FilesDsl(deltaClient: HttpClient)(implicit mat: Materializer, ec: ExecutionContext)
+    extends CirceUnmarshalling
+    with Matchers
+    with OptionValues
+    with ScalaFutures {
+
+  def uploadFile(
```

Comment:
The assert function being a parameter here makes this harder to compose, it's really better to just map to an assertion

Reply:
This might be better referred to as creating a file

Reply:
I agree - working with this style is painful. When I went down this rabbit hole it ended up touching pretty much all the integration tests 😓 this change is mostly just trying to group together storage-related things